src/script/components/MessagesList/Message/ContentMessage/asset/TextMessageRenderer/TextMessageRenderer.tsx

```diff
@@ -121,22 +119,36 @@ export const TextMessageRenderer: FC<TextMessageRendererProps> = ({
       forwardEvent(event.nativeEvent, 'mention', mentionMsgDetails);
     }
   };
+  const extraClasses = showFullText ? 'message-quote__text--full' : '';
 
   return (
-    // We will register the click event on the paragraph element and determine the type of the element clicked.
-    // This is because the paragraph element is fed with raw HTML and we cannot register the click event on the clickabled elements directly.
-    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
-    <p
-      ref={setContainerRef}
-      key={`${editedTimestamp}:${text}`}
-      onClick={handleInteraction}
-      onAuxClick={handleInteraction}
-      onKeyDown={handleInteraction}
-      onKeyUp={handleInteraction}
-      className={msgClass}
-      dangerouslySetInnerHTML={{__html: text}}
-      dir="auto"
-      {...props}
-    />
+    <>
+      {
+        // We will register the click event on the paragraph element and determine the type of the element clicked. //
+        //This is because the paragraph element is fed with raw HTML and we cannot register the click event on the
+        //clickabled elements directly.
+      }
+      {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
+      <p
+        ref={setContainerRef}
+        // We want to make sure that this element is re-rendered when the text changes (this will trigger the containerRef to be updated).
+        key={text}
+        onClick={handleInteraction}
+        onAuxClick={handleInteraction}
+        onKeyDown={handleInteraction}
+        onKeyUp={handleInteraction}
+        dangerouslySetInnerHTML={{__html: text}}
+        dir="auto"
+        className={`${className} ${extraClasses}`}
+        {...props}
+      />
+      {canShowMore && (
+        <ShowMoreButton
+          onClick={() => setShowFullText(!showFullText)}
```

Comment:
I'd rather move this function above render and use previous value when toggling state (state setters are not synchronous operations and might behave unexpectedly 😅 ):
```ts
const toggleShowFullText = () => {
  setShowFullText(prev => !prev)
}
```